### ISSUE TITLE
Add optimizer wrapper.

### ISSUE THREAD

**Issue by HesitantlyHuman** [HesitantlyHuman/autoclip#3]
It can be a hassle to add AutoClip into the training logic. It might be easier to give users the option to wrap their optimizer.
Something like:
```
optimizer = torch.optim.AdamW(model.parameters(), lr=1e-3, weight_decay=0.1)
clipper = autoclip.torch.QuantileClip(model.parameters(), clip_quantile=0.9)
optimizer = autoclip.torch.OptimizerWithClipping(optimizer=optimizer, clipper=clipper)
```
This would call the clipper and optimizer steps in the correct order behind the scenes, so the user wouldn't have to deal with any additional complexity.

**Comment by HesitantlyHuman** [comment]
This feature has been implemented in the [`optimizer-addons`](/HesitantlyHuman/autoclip/tree/optimizer-addons) branch.
While the above usage is technically possible, the encouraged pattern will be the following:
```
optimizer = torch.optim.AdamW(model.parameters())
optimizer = autoclip.torch.QuantileClip.as_optimizer(optimizer)
```
I'll close this issue when that branch has been merged and released.

**Comment by HesitantlyHuman** [comment]
Merged #5
